### DISCUSSION TITLE
How to delete a string from a map with strings?

### DISCUSSION THREAD

**Discussion by claudiosa** [vlang/v#14187] [General]
Hi everyone

I have a map such with type "map [string] [] string", which supports somehting such:

x  := { 'A': ['Brazil', 'Canada','Denmark']
    	  'B': ['France', 'Canada' ]
        }  
        
How to delete a 'Canada' from both keys in this map?
The Final expected result should be:

x  := { 'A': ['Brazil', 'Denmark']
    	  'B': ['France' ]
        }  

I tried to apply the filter but it does not support string types. Any guess? Any method and/or function to deal with maps with a set of strings?
Here x.delete('A') ... will delete the whole line or the key

Thanks

Claudio

**Comment by JalonSolov** [comment]
I don't know of a "short" way to do it, but...
```v
mut x := {
        'A': ['Brazil', 'Canada', 'Denmark']
        'B': ['France', 'Canada']
}

println('Before: $x')

for key, _ in x {
        i := x[key].index('Canada')
        if i >= 0 {
                x[key].delete(i)
        }
}

println('After: $x')
```

**Reply by claudio-sa** [reply]
Really I did not test the index like i := x[key].index('Canada'), to remove the by index, not by name. I tried the delete(nAME) directly with string, but it does not support yet. Thanks.

**Reply by JalonSolov** [reply]
Right.  `array.delete()` is documented taking an `i int` parameter... `delete deletes array element at index i.`

**Comment by Wajinn** [comment]
A slight variation, to see the values of the keys in the array.

```
for key_1, _ in x {
	for key_2, value in x[key_1] {
		println('values of A and B: $value')
		if value == 'Canada' {
			x[key_1].delete(key_2)
		}
	}
}
```

**Reply by claudio-sa** [reply]
@Wajinn , thanks also ... but in key_2, must be an integer !!! This part "for key_2, value in x[key_1]" is confusing for me yet.  Any way I will tried also.

**Reply by Wajinn** [reply]
key_1 and key_2 are both integers.  key_1 is just hiding (0,1) that is referring to A, B.  So to see the values it would be like x[0] (A), x[1] (B).  

key_2 is just a variable hiding the numbers (next set of keys) that come from (x[key_1]) or x[0], x[1]), which changes per each loop (0,1).  As is a double for-loop; x[0][0] (Brazil), x[0][1] (Canada), x[0][2] (Denmark). 

If you have a longer unfamiliar list, it can be helpful to see the values.  This way you can ponder about which ones you want to delete out.  Either way, both work, and can be a person's preference or a "good/fun to know".

**Reply by claudiosa** [reply]
Great, so, the maps remind a bidimensional matrix!

**Comment by claudiosa** [comment]
@JalonSolov  and @Wajinn 

Thanks for all, one more step was given. 
Thanks indeed.

The intermediate code is here: https://github.com/claudiosa/CCS/blob/master/v_programming_language/maps03_study.v
now, an algorithm for topology ordering, without BFS
